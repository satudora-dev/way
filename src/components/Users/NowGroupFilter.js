import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import FormGroup from '@material-ui/core/FormGroup';
import nowGroupList from '../nowGroupList';


const style = {
  tagStyle: {
    textAlign: "left",
    fontFamily: "Avenir",
    margin: "10px 30px",
    marginBottom: "40px",
  },
  btnStyle: {
    marginRight: "10px",
    marginBottom: "10px",
    backgroundColor: "#04B486",
    color: "white",
    textTransform: "none",
  },
}

export default class NowGroupFilter extends Component {

  filterNow(nowName) {
    if (nowName === 'all') {
      this.props.history.push('./users');
    }
    else {
      this.props.history.push(`./users?now=${nowName}`);
    }
  }

  render() {
    let nowGroup = ['all'];
    nowGroup = nowGroup.concat(nowGroupList);

    return (
      <div className="NowSwitch" style={style.tagStyle}>
        <FormGroup row>
          {nowGroup.map((item, i) => {
            return (
              <div key={i}>
                <Button
                  variant="contained"
                  style={style.btnStyle}
                  onClick={() => this.filterNow(item)} >
                  {item}
                </Button>
              </div>);
          })}
        </FormGroup>
      </div>
    );
  }
}
