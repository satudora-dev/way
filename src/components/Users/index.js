import React, { Component } from 'react';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Button from '@material-ui/core/Button';
import MenuAppBar from '../../components/MenuAppBar';


class Users extends Component {
  constructor(props){
    super(props);
    this.orderMODES={
      SORTED: 0,
      RANDOM: 1,
    };
    this.orderOptions=["RANDOM","PROJECT"];
    this.refineMODES={
      all: 0,
      position: 1,
      project: 2,
      tag: 3,
    };
    this.state={
      orderMode: this.orderMODES.SORTED,
      refineMode: this.refineMODES.all,
      refineKey: "",
      orderMenuOpen: false,
    };
  }

  componentWillMount(){
    const params= new URLSearchParams(this.props.location.search)
    //クエリの取得、なければ一覧
    if (params.get("position")) {
      const positionName = params.get("position")
      this.visibilityFilter = user => {
        return user.position === positionName;
      };
      this.setState({
        refineMode: this.refineMODES.position,
        refineKey: positionName,
      });
    }
    else if (params.get("project")) {
      const projectName = params.get("project")
      this.visibilityFilter = user => {
        return user.projects && user.projects.includes(projectName);
      };
      this.setState({
        refineMode: this.refineMODES.project,
        refineKey: projectName,
      });
    }
    else if (params.get("tag")) {
      const tagName = params.get("tag")
      this.visibilityFilter = user => {
        return user.tags && user.tags.includes(tagName);
      };
      this.setState({
        refineMode: this.refineMODES.tag,
        refineKey: tagName,
      });
    }
    else {
      this.visibilityFilter = user => {
        return true
      }
    }
  }


  toProfile(id){
    this.props.history.push(`/users/${id}`);
  }

  switchOrderMenu(on){
    this.setState({orderMenuOpen: on});
  }

  render() {
    const style = {
      iconstyle: {
        width: "128px",
        height: "128px",
        "object-fit": "cover",
        "border-radius": "50%",
        padding: "0px", //paddingだとその要素自体に余白つける
        //margin: "8px", //marginだとその要素の外側に余白つける
        "box-shadow": "0 4px 10px gray",
        cursor: "finger",
      },
      divstyle: {
        "text-align": "left",
        "padding": "10px 10px 10px",
      },
      btnstyle: {
        color: "white",
        "background-color": "#04B486",
      },
      iconbtnstyle: {
        "border-radius": "50%",
        padding: 0,
        margin: "5px"
      },
    };
    const users = this.props.users || {};


    const visibleUserKeys = Object.keys(users)
      .filter(key => this.visibilityFilter(users[key]))
    if(this.state.orderMode===this.orderMODES.SORTED){
      visibleUserKeys.sort((a,b)=>{//idを昇順にソートし、新規登録者を上に
        let aStr = a.toString();
        let bStr = b.toString();
        if(aStr > bStr) return -1;
        if(aStr < bStr) return 1;
        return 0;
      });
    }
    return (
      <div className="Users">
        <MenuAppBar/>

        {(()=>{
          if(this.state.refineKey!==""){
            return(
              <span>
                <br/>
                <Button variant="contained" style={style.btnstyle}>{this.state.refineKey}</Button>
                <hr />
              </span>
            );
          }
        })()}
        <div style={style.divstyle}>
        <Menu open={this.state.orderMenuOpen}>
          {this.orderOptions.map((mode,i)=>{
            return (
              <MenuItem key={i}
                onClick={()=>this.switchOrderMenu(false)}>
                {mode}
              </MenuItem>
            );
          })}
        </Menu>

        <div style={{"text-align":"center"}}>
        {visibleUserKeys.map( (key,i) =>{
          if(users[key]){
            return (
              <Button className="User" key={i} style={style.iconbtnstyle}
                onClick={()=>this.toProfile(key)}>
                <img src={users[key].icon} style={style.iconstyle} alt="failed loading..."/>
              </Button>
            );
          }
          else{
            return null;
          }
        })}
        </div>
        </div>
      </div>
    );
  }
}




export default Users;
