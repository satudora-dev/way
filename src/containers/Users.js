import Users from '../components/Users';
import { connect } from 'react-redux';



const mapStateToProps = ({auth, users}) => {
  const currentUserID = auth.currentUserID;
  if(currentUserID){
    return {
      currentUserID: currentUserID,
      hasOwnProfile: users[currentUserID] !== undefined,
      users: users
    }
  }
  else{
    return{
      currentUserID: null,
      hasOwnProfile: users[currentUserID] !== undefined,
      users: {}
    }
  }
}

export default connect(mapStateToProps,null)(Users)
