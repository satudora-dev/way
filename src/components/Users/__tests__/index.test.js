import React from 'react';
import Users from '../../Users';
import renderer from 'react-test-renderer';

test('Users snapshot', () => {
  const component = renderer.create(
    <Users />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
