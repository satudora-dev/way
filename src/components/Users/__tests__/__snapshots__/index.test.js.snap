// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users snapshot 1`] = `
<div
  className="User"
>
  <div>
    <header
      className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-7"
      style={
        Object {
          "height": "56px",
        }
      }
    >
      <div
        className="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38"
      >
        <div
          style={
            Object {
              "display": "table",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "display": "table-cell",
                "text-align": "center",
                "vertical-align": "middle",
              }
            }
          >
            <img
              alt="loading the way image..."
              src="/way.png"
              style={
                Object {
                  "height": "30px",
                }
              }
            />
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="NowSwitch"
    style={
      Object {
        "fontFamily": "Avenir",
        "margin": "10px 30px",
        "marginBottom": "40px",
        "textAlign": "left",
      }
    }
  >
    <div
      className="MuiFormGroup-root-41 MuiFormGroup-row-42"
    >
      <div>
        <button
          className="MuiButtonBase-root-69 MuiButton-root-43 MuiButton-contained-54 MuiButton-raised-57"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "#04B486",
              "color": "white",
              "marginBottom": "10px",
              "marginRight": "10px",
              "textTransform": "none",
            }
          }
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-44"
          >
            all
          </span>
          <span
            className="MuiTouchRipple-root-72"
          />
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root-69 MuiButton-root-43 MuiButton-contained-54 MuiButton-raised-57"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "#04B486",
              "color": "white",
              "marginBottom": "10px",
              "marginRight": "10px",
              "textTransform": "none",
            }
          }
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-44"
          >
            office
          </span>
          <span
            className="MuiTouchRipple-root-72"
          />
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root-69 MuiButton-root-43 MuiButton-contained-54 MuiButton-raised-57"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "#04B486",
              "color": "white",
              "marginBottom": "10px",
              "marginRight": "10px",
              "textTransform": "none",
            }
          }
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-44"
          >
            lunch
          </span>
          <span
            className="MuiTouchRipple-root-72"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "10px 10px 10px",
        "textAlign": "left",
      }
    }
  >
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
  </div>
</div>
`;
